### ISSUE TITLE
Install instructions give error at step to install redis on mac

### ISSUE THREAD

**Issue by AmandaBirmingham** [qiita-spots/qiita#3310]
The install instructions at

https://github.com/qiita-spots/qiita/blob/4492c3d53027a9387931b2449a705dd3b1c8f09f/INSTALL.md?plain=1#L119

give the command
`brew install homebrew/versions/redis28`
but Homebrew 4.1.4 (current version as of 08/11/2023) gives the error:
`Error: homebrew/versions was deprecated. This tap is now empty and all its contents were either deleted or migrated.`

@charles-cowart has indicated that it should be ok to use the latest version of redis instead (current up around 7.0), in which case the command in the install could change to:
`brew install redis`

**Comment by charles-cowart** [comment]
I think @antgonza should weigh in.

**Comment by AmandaBirmingham** [comment]
These may be the fastest issue responses I've ever seen :) :)

**Comment by antgonza** [comment]
FWIW, on my computer I'm using:
```
$ redis-server --version
Redis server v=5.0.3 sha=00000000:0 malloc=libc bits=64 build=499afec3908d6e99
```
and everything is working fine. Additionally, I didn't find in my email or notes any reason to keep using that older version.

I'm gonna issue a PR using the latest version and see what happens.

**Comment by antgonza** [comment]
Never mind, there are no restrictions in Qiita and the builds have been using higher version for both qiita and redbiom.

**Comment by antgonza** [comment]
Finally getting back to this.

Is it possible to let homebrew install whatever the newest version of redis is? If yes, would everyone agree that this could be a good solution for this issue? If not, what could it be a good solution?

**Comment by AmandaBirmingham** [comment]
Works for me :)

**Comment by charles-cowart** [comment]
> Finally getting back to this.
> 
> Is it possible to let homebrew install whatever the newest version of redis is? If yes, would everyone agree that this could be a good solution for this issue? If not, what could it be a good solution?

Agreed, it works for me as well. I believe 'brew install redis' will install the latest by default.

**Comment by wasade** [comment]
It assumes people use `brew` though which I think introduces a new developer expectation. Is there a reason the `conda-forge` build of `redis` doesn't work?

**Comment by wasade** [comment]
oh weird, `brew` is already part of the installation instructions? I didn't realize people actually used `brew`...

**Comment by charles-cowart** [comment]
@wasade it is also installable through conda (conda install redis), docker containers, and perhaps even other avenues. As long as the latest version can reliably be used then I don't think it matters so much how it's installed. Pinning the version to 2.x is what it makes it harder to install as it becomes unsupported on most of these installation pathways.

**Comment by wasade** [comment]
Yes, but we currently assume `conda` on deploy whereas we do not assume `docker` or `brew` (AFAIK).

Pins are hard and should be removed unless needed

**Comment by charles-cowart** [comment]
> Yes, but we currently assume `conda` on deploy whereas we do not assume `docker` or `brew` (AFAIK).
> 
> Pins are hard and should be removed unless needed

It sounds like we're all in agreement that this one can be removed then.

**Comment by wasade** [comment]
Possibly, were you able to perform the check discussed [here](https://github.com/qiita-spots/qiita/issues/3310#issuecomment-1675111722)?

**Comment by charles-cowart** [comment]
Confirmed the tests do pass using Redis 5.0.3 from the conda environment and using Redis 7.2.4 installed with homebrew. The number of messages sent using pubsub was changed to 1000 to shorten the test time; all other defaults were used. Not sure why two tests were skipped - do you know?

5.0.3:
(redbiom_test) ccowart@Applejack tests % python basic.py
..................ss.....
----------------------------------------------------------------------
Ran 25 tests in 0.034s

OK (skipped=2)
(redbiom_test) ccowart@Applejack tests % bash bench.sh
Testing on 127.0.0.1:7379 with 100 clients in parallel, for a total of 100,000 requests per benchmark.
PING: 58356.06 requests/sec.
SET(hello,world): 55115.43 requests/sec.
GET(hello): 56811.69 requests/sec.
INCR(hello): 57226.83 requests/sec.
LPUSH(hello,abc): 57086.33 requests/sec.
LRANGE(hello,0,10): 57221.14 requests/sec.
LRANGE(hello,0,100): 48426.13 requests/sec.
(redbiom_test) ccowart@Applejack tests % ./pubsub -n 1000
Received 1008 messages from 10 writers to 450 readers through 1 channels in 0.00 sec: received inf msg/sec
(redbiom_test) ccowart@Applejack tests % ./websocket 
After 0.06 sec: 3,329 messages sent, 10,000 received (100.00%). Instant rate: 169,765/sec, overall rate: 169,765/sec
(redbiom_test) ccowart@Applejack tests % bash curl-tests.sh 
Test 1/2: Large PUT upload: generating the key... OK
Sending a PUT request with a large payload... OK
Verifying the upload... OK

Test 2/2: Sending an OPTIONS request... OK
(redbiom_test) ccowart@Applejack tests % python limits.py 
test_huge_upload (__main__.TestHugeUrl) ... ok
test_huge_url (__main__.TestHugeUrl) ... ok

----------------------------------------------------------------------
Ran 2 tests in 0.204s

OK
(redbiom_test) ccowart@Applejack tests % 

7.2.4:
(redbiom_test) ccowart@Applejack tests % python basic.py
..................ss.....
----------------------------------------------------------------------
Ran 25 tests in 0.030s

OK (skipped=2)
(redbiom_test) ccowart@Applejack tests % bash bench.sh
Testing on 127.0.0.1:7379 with 100 clients in parallel, for a total of 100,000 requests per benchmark.
PING: 53388.40 requests/sec.
SET(hello,world): 53901.44 requests/sec.
GET(hello): 51155.26 requests/sec.
INCR(hello): 53898.65 requests/sec.
LPUSH(hello,abc): 55084.16 requests/sec.
LRANGE(hello,0,10): 54021.09 requests/sec.
LRANGE(hello,0,100): 43617.36 requests/sec.
(redbiom_test) ccowart@Applejack tests % ./pubsub -n 1000
Received 1008 messages from 10 writers to 450 readers through 1 channels in 0.00 sec: received inf msg/sec
(redbiom_test) ccowart@Applejack tests % ./websocket 
After 0.05 sec: 3,329 messages sent, 10,000 received (100.00%). Instant rate: 187,385/sec, overall rate: 187,385/sec
(redbiom_test) ccowart@Applejack tests % bash curl-tests.sh
Test 1/2: Large PUT upload: generating the key... OK
Sending a PUT request with a large payload... OK
Verifying the upload... OK

Test 2/2: Sending an OPTIONS request... OK
(redbiom_test) ccowart@Applejack tests % python limits.py
test_huge_upload (__main__.TestHugeUrl) ... ok
test_huge_url (__main__.TestHugeUrl) ... ok

----------------------------------------------------------------------
Ran 2 tests in 0.207s

OK

**Comment by charles-cowart** [comment]
> I think those are the redis tests, the earlier question was w.r.t. the redbiom test suite though?

doh! You're right, they were the webdis tests. Wrong directory.

**Comment by wasade** [comment]
No worries, thanks!

**Comment by charles-cowart** [comment]
[test.log](https://github.com/qiita-spots/qiita/files/13880568/test.log)
Confirmed redbiom tests completed successfully using Redis 7.2.4

**Comment by wasade** [comment]
Great, thank you!
